static_frame/core/frame.py

```diff
@@ -5968,6 +6013,27 @@ def to_latex(self,
                 )
         return repr(self.display(config))
 
+    def to_msgpack(self) -> 'msgpack':
+        '''
+        Return a msgpack.
+        '''
+        import msgpack
+        import msgpack_numpy
+        def cast_msgpack(input):
+            try:
+                data = msgpack.packb(input, default=msgpack_numpy.encode) #try coercing standard and numpy datatypes
+            except TypeError:
+                #This inserts a magic character with any arbitrary datatype that msgpack_numpy doesn't support
+                #TODO: Is this attempting to handle too much? Is there a better way?
+                clsname = '>'+input[0].__class__.__module__+'.'+input[0].__class__.__name__                
+                data = msgpack.packb([clsname]+[a.__str__() for a in input], use_bin_type=True) #else cast to string
+            return data
+        return cast_msgpack({
+            '_index' : cast_msgpack([index for index in self._index]),
+            '_columns' : cast_msgpack([column for column in self._columns]),
```

Comment:
For 1D `Index`, you want to get the NumPy array via `.values` instead of iterating through the `Index` and creating a new list. For 2D `IndexHierarchy`, you want to use the `._blocks` attribute, which is a `TypeBlocks` instance. You can get the list of contained arrays with `._blocks._blocks`.

We also need to capture the `name` of the `Index`.

Reply:
Ok, great. I figured I was missing some parts here, I'll work on understanding those use cases.

Reply:
now using .values for 1D arrays instead of repacking

Reply:
preserving index name now